tools/st2-self-check

```diff
@@ -47,31 +97,42 @@ TEST_ACTION_LIST=`st2 action list --pack=tests -w 90 | awk '{ print $2 }' | grep
 
 # Run all the tests
 for TEST in $TEST_ACTION_LIST
-    do
-        echo -n "Attempting Test ${TEST}..."
-        TEST_OUTPUT=`st2 run ${TEST} token=${ST2_AUTH_TOKEN} -j`
-        CHECK_STATUS=`echo ${TEST_OUTPUT} | grep '"status": "succeeded"'`
-        if [ $? -ne 0 ]; then
-            echo -e "ERROR!"
-            ((ERRORS++))
-        else
-            echo "OK!"
-        fi
-    done
+do
+    # If -w option is not set, skip Windows related tests
+    # because smbclient is not installed by default.
+    if [ ${RUN_WINDOWS_TESTS} = "false" ] && [ ${TEST} = "tests.test_windows_runners" ]; then
+        echo "Skipping ${TEST}..."
+        continue
+    fi
+
+    echo -n "Attempting Test ${TEST}..."
+    TEST_OUTPUT=`st2 run ${TEST} protocol=${PROTOCOL} token=${ST2_AUTH_TOKEN} -j`
+    CHECK_STATUS=`echo ${TEST_OUTPUT} | grep '"status": "failed"'`
+    if [ $? -ne 1 ]; then
+        echo -e "ERROR!"
+        ((ERRORS++))
+    else
+        echo "OK!"
+    fi
+done
 
 echo -n "Attempting Example examples.mistral_examples..."
 MISTRAL_EXAMPLES=`st2 run examples.mistral_examples -j`
-CHECK_STATUS=`echo ${MISTRAL_EXAMPLES} | grep '"status": "succeeded"'`
-if [ $? -ne 0 ]; then
+CHECK_STATUS=`echo ${MISTRAL_EXAMPLES} | grep '"status": "failed"'`
```

Comment:
Because during some runs, tests failed but since action chain and mistral lists all status, some of the task status gets in the way.  I just find out failed doesn't work either because we expect some tasks to fail...